mesonbuild/build.py

```diff
@@ -1025,6 +1026,13 @@ def get_using_msvc(self):
     def is_linkable_target(self):
         return False
 
+    def check_module_linking(self):
+        '''
+        Warn if shared modules are linked with target: (link_with) #2865
+        '''
+        for link_target in self.link_targets:
+            if isinstance(link_target, SharedModule):
+                mlog.warning('''Linking shared modules to targets is not recommended''')
```

Comment:
This should be a bit more explicit about why. Something like "Warning: target links against shared modules. This is not recommended as it can lead to undefined behaviour on some platforms."

To avoid spamming the user too much, the function should `return` after printing the message. There is no point in repeating the same warning multiple times.

Also, there is no need to use triple quotes on single line messages. They are only visual noise.